### ISSUE TITLE
val directory in pix2pix

### ISSUE THREAD

**Issue by HJLYU0519** [junyanz/pytorch-CycleGAN-and-pix2pix#532]
I want to run pix2pix code on my own dataset. And I found this sentence in tips:
        A and B should each have their own subdirectories train, val, test, etc.
My question is:  What kind of image is in 'val' folder?
     I know I need put training images in 'train' folder and testing images in 'test' folder. What about the 'val' folder?

**Comment by junyanz** [comment]
You can just have `train` and `test` directories.  Sometimes people use `val` for validation set. But it is optional.

**Comment by HJLYU0519** [comment]
Ｍany thanks for your kind reply which has helped me solve my problems.

After  finishing the training process, I found another question:
the learning rate was always the same with the starting value (0.002), even if I have not made any changes to lr_policy (ie set lr_policy as linear). 
Could you tell me why,please？
Thanks for your kind help in advance！

**Comment by HJLYU0519** [comment]
Sorry, the accurate value of the learning rate was 0.0002000. But this is not a big deal for my question.
I am wondering why the learning rate didn't change during training process.
Thanks again!

**Comment by junyanz** [comment]
It will be the same for the first 100 epochs and linearly decreases to 0 in the next 100 epochs.

**Comment by HJLYU0519** [comment]
I have checked the get_scheduler function in network.py. I knew lr would linearly decrease to 1 after first 100 epochs. But the training process is different from what was expected.
Again, I have not made any changes to lr_policy (ie set lr_policy as linear).

**Comment by HJLYU0519** [comment]
I think the epoch in get_scheduler function for linear policy could not  change during training process.
        def lambda_rule(epoch):
            lr_l = 1.0 - max(0, epoch + opt.epoch_count - opt.niter) / float(opt.niter_decay + 1)
since the epoch would not change, lr_l would keep the same value with the starting value

**Comment by HJLYU0519** [comment]
When I delete self.metric from  update_learning_rate(self) function in base_model.py,  he learning rate change as what we expected.
I know that adding self.metric is to solve other problems. But after I deleted it, it works on my code.

**Comment by junyanz** [comment]
I have fixed this problem with the latest commit. Please check out the code again.
